host/src/att.rs

```diff
@@ -28,70 +31,52 @@ pub(crate) const ATT_READ_BLOB_REQ: u8 = 0x0c;
 pub(crate) const ATT_READ_BLOB_RSP: u8 = 0x0d;
 pub(crate) const ATT_HANDLE_VALUE_NTF: u8 = 0x1b;
 
+/// Attribute Error Code
+///
+/// This enum type describes the `ATT_ERROR_RSP` PDU from the Bluetooth Core Specification
+/// Version 6.0 | Vol 3, Part F (page 1491)
 #[cfg_attr(feature = "defmt", derive(defmt::Format))]
-#[derive(Debug, PartialEq, Clone, Copy)]
+#[derive(Debug, PartialEq, Clone, Copy, TryFromPrimitive, Error)]
 #[repr(u8)]
 pub enum AttErrorCode {
```

Comment:
I think we should consider making this a newtype wrapping `u8` with associated constants, rather than an enum. When a client sends a request to a remote server that server can theoretically return any error code it wants, regardless of the spec. For example, if it's a server that supports a newer version of the spec than trouble it could return a code unknown to trouble.

Reply:
You make a good point, there is actually a range reserved for application-defined error codes which this wouldn't be able to handle. I think it's beyond the scope of this PR as the logic here hasn't changed, but I'll raise it as an issue to be addressed.

Reply:
https://github.com/embassy-rs/trouble/issues/161